source-code/app/src/main/java/org/buildmlearn/toolkit/fragment/DraftsFragment.java

```diff
@@ -397,4 +408,22 @@ private void deleteAllDrafts() {
     private void restoreSelectedView() {
         restoreColorScheme();
     }
+
+    @Override
+    public void onBackPressed() {
+
+        if (mAdapter.selectedPositionsSize() != 0){
+            for(int i=0;i<mAdapter.getCount();i++)
+                if(mAdapter.isPositionSelected(i)) {
+                    mListView.getChildAt(i).setBackgroundColor(0);
+                    mAdapter.removeSelectedPosition(i);
+                }
+            restoreColorScheme();
+        }
+        else{
+            FragmentManager fragmentManager = getFragmentManager();
+            fragmentManager.beginTransaction().setTransition(FragmentTransaction.TRANSIT_FRAGMENT_FADE)
```

Comment:
Whenever we press back, we should not be performing a new fragment transaction, rather will pop the last fragment from the Stack.